pkg/chunk/cache/fifo_cache.go

```diff
@@ -162,12 +173,22 @@ func (c *FifoCache) Store(ctx context.Context, keys []string, bufs [][]byte) {
 
 // Stop implements Cache.
 func (c *FifoCache) Stop() {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	// update metrics
+	c.entriesEvicted.Add(float64(len(c.entries)))
+	c.entriesCurrent.Set(float64(0))
+	c.memoryBytes.Set(float64(0))
+
+	c.entries = nil
```

Comment:
To cover the case someone accidentally calls the cache functions after `Stop()`
```suggestion
	c.entries = make(map[string]*cacheEntry)
```